modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -340,3 +340,13 @@ fun <A, B> B?.rightIfNotNull(default: () -> A): Either<A, B> = when (this) {
   null -> Either.Left(default())
   else -> Either.Right(this)
 }
+
+/**
+ * Convenience extension for [Collection<Either<*, A>>] to get a [Collection<A>] with only the Either.rights.
+ *
+ * Example:
+ * ```
+ * listOf("value".right(), "other".left()).collect() // ["value"]
+ * ```
+ */
+fun <A> Collection<Either<*, A>>.collect(): Collection<A> where A : Any = mapNotNull { it.orNull() }
```

Comment:
This can be generalized for all `Foldable` data types not just either. In Arrow 0.8.0 all extensions declared in Foldable will be projected as extension functions over the datatype. We already have also a version of `Collect` elsewhere which uses `PartialFunction` that could potentially be moved to Foldable as well. In general, we prefer in Arrow to not provide concrete extension functions for a particular datatype if the same functionality can be expressed polymorphically as part of a type class that many data types provide instances for.

I would encourage you to explore moving collect with `PartialFunction` inside the `Foldable` interface